From 7b5af8c6076e259393d56ad593ca9b2f36e670f3 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Thu, 20 Jan 2022 19:29:29 +0200
Subject: [PATCH] imx8mp: compulab: Add usbdev device tree

Due to the fact that otg support is not implemented by the SW yet, this
temporary solution is provided. Make use of a -usbdev.dtb in order to
get a USB peripheral support on the USB-type-C port (USB1).

Signed-off-by: Valentin Raevsky <valentin@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/Makefile         |  2 ++
 .../dts/compulab/som-imx8m-plus-usbdev.dts    | 27 +++++++++++++++++++
 .../dts/compulab/ucm-imx8m-plus-usbdev.dts    | 27 +++++++++++++++++++
 3 files changed, 56 insertions(+)
 create mode 100644 arch/arm64/boot/dts/compulab/som-imx8m-plus-usbdev.dts
 create mode 100644 arch/arm64/boot/dts/compulab/ucm-imx8m-plus-usbdev.dts

diff --git a/arch/arm64/boot/dts/compulab/Makefile b/arch/arm64/boot/dts/compulab/Makefile
index 64d25671aa50..a03513f67762 100644
--- a/arch/arm64/boot/dts/compulab/Makefile
+++ b/arch/arm64/boot/dts/compulab/Makefile
@@ -6,6 +6,7 @@ dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus_mipi-csi1-imx219.dtb
 dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus-mipi.dtb
 dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus-lvds.dtb
 dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus-hdmi.dtb
+dtb-$(CONFIG_ARCH_MXC) += ucm-imx8m-plus-usbdev.dtb
 
 dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus.dtb
 dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus-thermal.dtb
@@ -14,6 +15,7 @@ dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus_mipi-csi2.dtb
 dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus-mipi.dtb
 dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus-lvds.dtb
 dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus-hdmi.dtb
+dtb-$(CONFIG_ARCH_MXC) += som-imx8m-plus-usbdev.dtb
 
 always-y       := $(dtb-y)
 subdir-y       := $(dts-dirs)
diff --git a/arch/arm64/boot/dts/compulab/som-imx8m-plus-usbdev.dts b/arch/arm64/boot/dts/compulab/som-imx8m-plus-usbdev.dts
new file mode 100644
index 000000000000..83c38c3812aa
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/som-imx8m-plus-usbdev.dts
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 CompuLab
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "som-imx8m-plus.dts"
+
+/ {
+	model = "CompuLab SOM-iMX8M-Plus (usbdev)";
+	compatible = "compulab,som-imx8m-plus", "fsl,imx8mp";
+};
+
+&usb_dwc3_0 {
+	dr_mode = "peripheral";
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-usbdev.dts b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-usbdev.dts
new file mode 100644
index 000000000000..52555e2a54a6
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-usbdev.dts
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2021 CompuLab
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "ucm-imx8m-plus.dts"
+
+/ {
+	model = "CompuLab UCM-iMX8M-Plus (usbdev)";
+	compatible = "compulab,ucm-imx8m-plus", "fsl,imx8mp";
+};
+
+&usb_dwc3_0 {
+	dr_mode = "peripheral";
+	status = "okay";
+};
-- 
2.17.1

